### PR TITLE
RDM-12049: Apply the security classification filters based on the superset of the <classification>(s)

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-12049

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.AccessProfile;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol.CaseDataAccessControl;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class AttributeBasedSecurityClassificationServiceTest {
```
Comment (danlysiak) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationServiceTest.java:19]: Maybe worth adding a comment to `SecurityClassificationServiceTest` to remind whoever ends up removing that in the future to move the tests into here

**Review by danlysiak — COMMENTED** [review]
Looks good so far 👍

File: src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationService.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.service.AccessControl;
+import uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol.CaseDataAccessControl;
+
+import static java.util.Comparator.comparingInt;
+
+@Service
+@Primary
+@ConditionalOnProperty(name = "enable-attribute-based-access-control", havingValue = "true")
```
[the remark this answers is not in the document]
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationService.java]: done

File: src/test/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationServiceTest.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply src/test/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedSecurityClassificationServiceTest.java:20]: added test cases related to accesscontrol logic

**Pull request merged**